linalg/solvers.cpp

```diff
@@ -338,18 +333,20 @@ void CGSolver::Mult(const Vector &b, Vector &x) const
    oper->Mult(d, z);  // z = A d
    den = Dot(z, d);
    MFEM_ASSERT(IsFinite(den), "den = " << den);
-
-   if (print_level >= 0 && den < 0.0)
-   {
-      mfem::out << "Negative denominator in step 0 of PCG: " << den << '\n';
-   }
-
-   if (den == 0.0)
+   if (den <= 0.0)
    {
-      converged = 0;
-      final_iter = 0;
-      final_norm = sqrt(nom);
-      return;
+      if (Dot(d, d) > 0.0 && print_level >= 0)
```

Comment:
To avoid the extra computation when `print_level` is set to `0`
```suggestion
      if (print_level >= 0 && Dot(d, d) > 0.0)
```

Reply:
That does not work in parallel where `print_level` may be different on different processors (that's how we get print output from rank 0 only). The original code was causing deadlock in parallel. Anyway, this code is executed only when an error-like condition is encountered, so performance is not crucial here.